source/cppfs/include/cppfs/null/NullFileWatcher.h

```diff
@@ -0,0 +1,65 @@
+
+#pragma once
+
+
+#include <memory>
+
+#include <cppfs/AbstractFileWatcherBackend.h>
+
+
+namespace cppfs
+{
+
+
+class AbstractFileSystem;
+
+
+/**
+*  @brief
+*    Empty implementation of the file watcher interface
+*
+*  @remarks
+*    This class is used for file systems that do not support
+*    file system watchers, such as remote file systems.
+*/
+class CPPFS_API NullFileWatcher : public AbstractFileWatcherBackend
```

Comment:
I used it in some of my older projects with the private implementation pattern (which we also use here). It makes it simpler because one can just assume that the implementation object is always valid and doesn't have to check it in each function. But you're right, we're checking m_backend in all functions to far, so maybe we should stick to that pattern (use nullptr instead of null-implementation)